eng/scripts/Get-BinarySizes.ps1

```diff
@@ -41,13 +41,16 @@ function setEnvVar($key, $value) {
 }
 
 function getTargetOs {
-    if ($OsVMImage.StartsWith('macOS')) {
+    if ($OsVMImage.StartsWith('macOS', $true, (Get-Culture).InvariantCulture)) {
```

Comment:
I assume #3900 will go in first?

Reply:
They're going into two different branches - #3900 is going into main, this is going into /feature/websockets. Basically I ran into #3899 while I was working on this fix and I fixed it here. I realized this was affecting our nightly builds so I pulled some pieces of this PR into main (the reformatting of the platform matrix JSON files and this fix).

I pulled the platform matrix reformatting to make it easier to ultimately merge the /feature/websockets branch with main.